src/components/Tooltip/Tooltip.js

```diff
@@ -1,32 +1,72 @@
-import React, { Fragment } from "react";
-import PropType from "prop-types";
+import React, { Fragment, useState } from "react";
+import PropTypes from "prop-types";
 
+/* eslint-disable react/prop-types */
 const DefaultTooltipActivator = () => {
     return (<Fragment>i</Fragment>);
 };
   
 const Tooltip = (props) => {
     const {
-        tooltipActivator,
-        children
+        TooltipActivator,
+        activatorAction,
+        direction,
+        children,
+        ...restProps
     } = props;
 
+    const [actionClassName, setActionClassName] = useState("");
+    const [isOpen,setIsOpen] = useState(false);
+
+    let dirClassName = "RCB-tooltip-right";
+    if (direction === "bottom") {
+        dirClassName = "RCB-tooltip-bottom"
+    }
+
+    const showToolTipClick = (e) => {
+        e.stopPropagation();
+        setIsOpen(!isOpen);
+        setActionClassName("RCB-tooltip-click");
+    }
+
+    const showToolTipHover = () => {
+        setActionClassName("RCB-tooltip-hover");
+        setIsOpen(true);
+    }
+
+    const hideToolTip = () => {
+        setIsOpen(false);
+    }
+    
     return (<div className="RCB-tooltip">
-        <div className="RCB-tooltip-btn">
-            {tooltipActivator}
-        </div>
-        <div className="RCB-tooltip-body">
+        { activatorAction === "click" ? 
+            (
+                <div className="RCB-tooltip-btn" onClick={showToolTipClick}>
+                    <TooltipActivator {...restProps}></TooltipActivator>
+                </div>
+            ) : 
+            (
+                <div className="RCB-tooltip-btn" onMouseEnter={showToolTipHover} onMouseLeave={hideToolTip}>
+                    <TooltipActivator {...restProps}></TooltipActivator>
+                </div>
+            )
```

Comment:
you can handle this by putting an if / else for the props instead so as to reduce the redundant JSX

```
const eventProps = (activatorAction === "click")  ? {onClick: showToolTipClick} : {onMouseEnter: showToolTipHover, onMouseLeave: hideToolTip};
                <div className="RCB-tooltip-btn" {...eventProps}>
                    <TooltipActivator {...restProps}></TooltipActivator>
                </div>
```

Reply:
DOne